src/OpenLoco/src/GameCommands/RaiseLowerLand.cpp

```diff
@@ -0,0 +1,485 @@
+
+    // 0x00468651
+    // TODO: move to TileManager?
+    static uint32_t adjustSurfaceHeight(World::Pos2 pos, SmallZ targetBaseZ, uint8_t slopeFlags, std::set<World::Pos3, LessThanPos3>& removedBuildings, uint8_t flags)
+    {
+        if (!validCoords(pos))
+        {
+            setErrorText(StringIds::off_edge_of_map);
+            return FAILURE;
+        }
+
+        if (targetBaseZ < 4)
+        {
+            setErrorText(StringIds::error_too_low);
+            return FAILURE;
+        }
+        if (targetBaseZ > 160
+            || (targetBaseZ == 160 && (slopeFlags & 0x1F) == 0)
+            || (targetBaseZ == 156 && (slopeFlags & 0x10) == 0))
+        {
+            setErrorText(StringIds::error_too_high);
+            return FAILURE;
+        }
+
+        currency32_t totalCost = 0;
+
+        if (flags & Flags::apply)
+        {
+            TileManager::removeSurfaceIndustry(pos);
+            TileManager::setTerrainStyleAsCleared(pos);
+
+            auto clearHeight = TileManager::getHeight(pos).landHeight;
+            TileManager::removeAllWallsOnTile(toTileSpace(pos), clearHeight);
+        }
+
+        // Compute cost of landscape operation
+        auto tile = TileManager::get(pos);
+        auto* surface = tile.surface();
+        auto landObj = ObjectManager::get<LandObject>(surface->terrain());
+        totalCost = totalCost + Economy::getInflationAdjustedCost(landObj->costFactor, landObj->costIndex, 10);
+
+        // If surface is in use as a water(route), and has water, ensure we don't remove it
+        if (surface->hasType6Flag() && surface->water() > 0)
+        {
+            auto water = surface->water() * 2 - kMicroToSmallZStep;
+            auto testHeight = targetBaseZ;
+            if (slopeFlags & 0x1F)
+            {
+                testHeight += kSmallZStep;
+                if (slopeFlags & SurfaceSlope::doubleHeight)
+                {
+                    testHeight += kSmallZStep;
+                }
+            }
+
+            if (water < testHeight)
+            {
+                setErrorText(StringIds::water_channel_currently_needed_by_ships);
+                return FAILURE;
+            }
+        }
+
+        // Bind our local vars to the tile clear function
+        auto clearFunc = [pos, &removedBuildings, flags, &totalCost](TileElement& el) {
+            return TileClearance::tileClearFunction(el, pos, removedBuildings, flags, totalCost);
+        };
+
+        QuarterTile qt(0xF, 0);
+        auto tileHeight = TileManager::getHeight(pos);
+        if (!TileClearance::applyClearAtAllHeights(pos, tileHeight.landHeight / 4, tileHeight.landHeight / 4, qt, clearFunc))
+        {
+            return GameCommands::FAILURE;
+        }
+
+        // Check bridge requirements for track and road elements
+        const auto tileIt = TileManager::get(pos);
+        for (auto& el : tileIt)
+        {
+            if (!(el.type() == ElementType::track || el.type() == ElementType::road))
```

Comment:
i would just do `auto* trackEl = el.as<TrackElement>(); auto* roadEl = el.as<RoadElement>();` at the start then you don't need to check the type 3 times.